File: frontend/dorm-app/src/lib/types/inspection.ts
export type InspectionStatus = "Scheduled" | "Active" | "Completed";

export type RoomInspectionStatus =
  | "Pending"
  | "Confirmed"
  | "NotConfirmed"
  | "NoAccess";

export interface RoomInspection {
  id: string;
  roomNumber: string;
  floor: string;
  building: string;
  status: RoomInspectionStatus;
  comment?: string;
}

export interface Inspection {
  id: string;
  name: string;
  type: string;
  startDate: Date;
  status: InspectionStatus;
  rooms: RoomInspection[];
} 

export interface ShortInspection {
  id: string;
  name: string;
  type: string;
  startDate: Date;
  status: InspectionStatus;
  roomsCount: number;
  pendingRoomsCount: number;
  confirmedRoomsCount: number;
  notConfirmedRoomsCount: number;
  noAccessRoomsCount: number;
}

File: Services/Inspections/Inspections.API/Entities/Inspection.cs
namespace Inspections.API.Entities
{
    public class Inspection
    {
        public Guid Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Type { get; set; } = string.Empty;

        public DateTime StartDate { get; set; }

        public InspectionStatus Status { get; set; }

        public ICollection<RoomInspection> Rooms { get; set; } = new List<RoomInspection>();
    }
}